contrib/scalar/scalar.c

```diff
@@ -1005,6 +1005,51 @@ static void dir_file_stats(struct strbuf *buf, const char *path)
 				    e_stat.st_size);
 }
 
+static int count_files(char *path)
+{
+	DIR *dir = opendir(path);
+	struct dirent *e;
+	int count = 0;
+
+	if (!dir)
+		return 0;
+
+	while ((e = readdir(dir)) != NULL)
+		if (!is_dot_or_dotdot(e->d_name) && e->d_type == DT_REG)
+			count++;
+
+	return count;
+}
+
+#define IS_HEX_DIGIT(c) (('0' <= (c) && (c) <= '9') || \
+			 ('A' <= (c) && (c) <= 'F') || \
+			 ('a' <= (c) && (c) <= 'f'))
+
+static void loose_objs_stats(struct strbuf *buf, const char *path)
+{
+	DIR *dir = opendir(path);
+	struct dirent *e;
+	int count;
+	int total = 0;
+
+	if (!dir)
+		return;
+
+	strbuf_addstr(buf, "Object directory stats for ");
+	strbuf_add_absolute_path(buf, path);
+	strbuf_addstr(buf, ":\n");
+
+	while ((e = readdir(dir)) != NULL)
+		if (!is_dot_or_dotdot(e->d_name) &&
+		    e->d_type == DT_DIR && strlen(e->d_name) == 2 &&
+		    IS_HEX_DIGIT(e->d_name[0]) && IS_HEX_DIGIT(e->d_name[1])) {
```

Comment:
Could we use something like this instead?

```c
unsigned char c;

if (... !hex_to_bytes(&c, e->d_name, 2))
```

This would make `IS_HEX_DIGIT()` unnecessary.

Reply:
We could do, and that would work.

Do we care about any downsides of calling `hex_to_bytes` in a loop over the directory names?? vs the inline constant comparisons?? I guess the `diagnose` command isn't exactly perf-critical.

(I was surprised there wasn't any existing `IS_HEX_DIGIT`-like construct.. but then maybe I didn't look hard enough to find `hex_to_bytes`?)

Reply:
Moved to `hex_to_bytes(&c, e->d_name, 1)` (note that it's 1 byte `==` 2 characters).